### ISSUE TITLE
Remove Mojolicious (Perl) Server heading in Production

### ISSUE THREAD

**Issue by srchulo** [mojolicious/mojo#1339]
In [Mojo::Server::Daemon](https://metacpan.org/pod/Mojo::Server::Daemon) the [Server](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server) header is set to "Mojolicious (Perl)":

https://github.com/mojolicious/mojo/blob/543f118ed7ee3d9d15431b2f5e335fa116a8e193/lib/Mojo/Server/Daemon.pm#L80

I wonder if in production mode this should be removed, as it could make it slightly easier for someone to abuse this knowledge if there was a Mojolicious exploit, and I don't think it offers up much to the user?

**Comment by shadowcat-mst** [comment]
You can remove it in your application code if you want to.

**Comment by srchulo** [comment]
Yeah, I just thought it could be good generally as a security measure for all Mojolicious users, and I thought it was a sensible default. If the Mojolicious maintainers don't feel the same, feel free to close this and I will just write a plugin to do it.

**Comment by kraih** [comment]
And there are other tricks for identifying a Mojolicious application. Like requesting the [failraptor](https://mojolicious.org/mojo/failraptor.png).

**Comment by srchulo** [comment]
That makes sense, but the "Server" header makes it very obvious if someone is looking for that information. If they do not know that it is a Mojolicious application, they may not know to look for something like the failraptor. I don't think removing the Server header makes an application way more secure, but I think there is potential benefit, but I don't see the benefit of including it in a production environment?

**Comment by Tekki** [comment]
[rfc2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.38): 'Server implementors are encouraged to make this field a configurable option.' Not to remove it completely.

**Comment by srchulo** [comment]
I'll post the full quote, as I think it's relevant:

> Note: Revealing the specific software version of the server might
>       allow the server machine to become more vulnerable to attacks
>       against software that is known to contain security holes. Server
>       implementors are encouraged to make this field a configurable
>       option.

There they are talking about the software version of the server, but I think it still applies to the server in general.

**Comment by jhthorsen** [comment]
:-1 from me.

**Comment by srchulo** [comment]
@jhthorsen could you explain what the downside is or why you think it shouldn't be added? Do you not agree with the security perspective?

**Comment by srchulo** [comment]
And could you explain what the upside is?

**Comment by kraih** [comment]
Removing the `Server` header is already an [example](https://mojolicious.org/perldoc/Mojolicious/Guides/Routing#Hooks) in the guides.

**Comment by jhthorsen** [comment]
I do not agree on the security perspective, and I think it’s nice advertisement.

Nginx serve with a default header set all over the place. I think if that’s still an ok default for them, then I think it’s ok for us as well.

**Comment by srchulo** [comment]
Okay, well I will just use the after_dispatch hook from the guides.

Thanks for the consideration!

**Comment by kraih** [comment]
@srchulo If just finding a solution to your problem was the goal here, please use our official support channels in the future. https://mojolicious.org/perldoc#SUPPORT

**Comment by srchulo** [comment]
I wasn't looking for a solution, I had planned to create a Mojolicious::Plugin with that after_dispatch hook, but I thought it would be a generally useful feature for everyone using Mojolicious (I was surprised when I saw that it was still there in production). I closed because it seemed like the consensus was that it wasn't the right decision.

In the future I'll use those support channels if I do have a support question.
